fms2_io/fms_netcdf_domain_io.F90

```diff
@@ -546,34 +546,28 @@ subroutine save_domain_restart(fileobj, unlim_dim_level)
   if (.not. fileobj%is_restart) then
     call error("file "//trim(fileobj%path)//" is not a restart file.")
   endif
+
+! Calculate the checksum and write it out
   do i = 1, fileobj%num_restart_vars
     if (associated(fileobj%restart_vars(i)%data0d)) then
-      call domain_write_0d(fileobj, fileobj%restart_vars(i)%varname, &
-                           fileobj%restart_vars(i)%data0d, unlim_dim_level=unlim_dim_level)
+       cycle
     elseif (associated(fileobj%restart_vars(i)%data1d)) then
-      call domain_write_1d(fileobj, fileobj%restart_vars(i)%varname, &
-                           fileobj%restart_vars(i)%data1d, unlim_dim_level=unlim_dim_level)
+       cycle
```

Comment:
Can these be combined or removed since nothing is happening?

Reply:
They can't be removed because, if you have a 0d or 1d variable, it will end up in: 
```
    else
       call error("this branch should not be reached.")
     endif
```

Reply:
The `if`s should be combined then instead of two different `if`s.